virttest/migration_template.py

```diff
@@ -0,0 +1,563 @@
+        """
+        Set up native encryption migration env
+        """
+        conn_args = {'server_ip': self.params['migrate_dest_host'],
+                     'server_cn': self.params['migrate_dest_host_cn'],
+                     'server_pwd': self.params['migrate_dest_pwd'],
+                     'client_ip': self.params['migrate_source_host'],
+                     'client_cn': self.params['migrate_source_host_cn'],
+                     'client_pwd': self.params['migrate_source_pwd']}
+
+        back_conn_args = {'client_ip': self.params['migrate_dest_host'],
+                          'client_cn': self.params['migrate_dest_host_cn'],
+                          'client_pwd': self.params['migrate_dest_pwd'],
+                          'server_ip': self.params['migrate_source_host'],
+                          'server_cn': self.params['migrate_source_host_cn'],
+                          'server_pwd': self.params['migrate_source_pwd']}
+
+        if self.params['virsh_migrate_options'].count('--tls'):
+            logging.debug("Setup qemu tls env")
+            qemu_tls_args = dict(conn_args)
+            qemu_tls_args_ = {'custom_pki_path': '/etc/pki/qemu',
+                              'qemu_tls': 'yes'}
+            qemu_tls_args.update(qemu_tls_args_)
+            qemu_tls_obj = TLSConnection(qemu_tls_args)
+            qemu_tls_obj.conn_setup()
+            qemu_tls_obj.auto_recover = True
+            self.objs_list.append(qemu_tls_obj)
+            if self.params['migrate_vm_back'] == 'yes':
+                logging.debug("Setup qemu tls env for reverse migration")
+                back_qemu_tls_args = dict(back_conn_args)
+                back_qemu_tls_args_ = {'ca_cakey_path': qemu_tls_args.get('custom_pki_path'),
+                                       'scp_new_cacert': 'no'}
+                back_qemu_tls_args.update(qemu_tls_args_)
+                back_qemu_tls_args.update(back_qemu_tls_args_)
+                back_qemu_tls_obj = TLSConnection(back_qemu_tls_args)
+                back_qemu_tls_obj.conn_setup()
+                back_qemu_tls_obj.auto_recover = True
+                self.objs_list.append(back_qemu_tls_obj)
+
+    def cleanup(self):
+        """Cleanup env"""
+        logging.debug("Start to clean up env")
+        # Shutdown vms
+        for vm in self.migrate_vms:
+            vm.destroy()
+
+        # Recover source vm defination (just in case).
+        logging.info("Recover vm defination on source")
+        for backup in self.vm_xml_backup:
+            backup.define()
+
+        # Clean up ssh, tcp, tls test env
+        if self.objs_list and len(self.objs_list) > 0:
+            logging.debug("Clean up test env: ssh, tcp, tls, etc")
+            self.objs_list.reverse()
+            for obj in self.objs_list:
+                obj.__del__()
+
+        # Disable libvirtd remote connection port
+        if self.remote_port:
+            remote_dict = {'server_ip': self.params['remote_ip'],
```

Comment:
This variable also can be defined as global variable or instant variable for reuse.

Reply:
Wrap it into new method: _open_libvirtd_remote_port_in_iptables()